librz/bin/pdb/pdb.c

```diff
@@ -1,1433 +1,589 @@
- * \param size Size of the structure/class
- * \param members List of members
- * \param printf Print function
- */
-static void print_struct(const char *name, const int size, const RzList *members, PrintfCallback printf) {
-	rz_return_if_fail(name && printf);
-	printf("struct %s { // size 0x%x\n", name, size);
-
-	RzListIter *member_iter;
-	STypeInfo *type_info;
-	rz_list_foreach (members, member_iter, type_info) {
-		switch (type_info->leaf_type) {
-		case eLF_MEMBER:
-		case eLF_NESTTYPE:
-		case eLF_METHOD:
-		case eLF_ONEMETHOD: {
-			char *member_name = NULL;
-			if (type_info->get_name) {
-				member_name = type_info->get_name(type_info);
-			}
-			ut64 offset = 0;
-			if (type_info->get_val) {
-				offset = type_info->get_val(type_info);
-			}
-			char *type_name = NULL;
-			if (type_info->get_print_type) {
-				type_info->get_print_type(type_info, &type_name);
-			}
-			printf("  %s %s; // offset +0x%" PFMT64x "\n", type_name, member_name, offset);
-			RZ_FREE(type_name);
 		}
 		default:
 			break;
 		}
 	}
-	printf("};\n");
-}
-
-/**
- * \brief Prints out union leaf type
- * 
- * \param name Name of the union
- * \param size Size of the union
- * \param members List of members
- * \param printf Print function
- */
-static void print_union(const char *name, const int size, const RzList *members, PrintfCallback printf) {
-	rz_return_if_fail(name && printf);
-	printf("union %s { // size 0x%x\n", name, size);
-
-	RzListIter *member_iter;
-	STypeInfo *type_info;
-	rz_list_foreach (members, member_iter, type_info) {
-		char *member_name = NULL;
-		if (type_info->get_name) {
-			member_name = type_info->get_name(type_info);
-		}
-		ut64 offset = 0;
-		if (type_info->get_val) {
-			offset = type_info->get_val(type_info);
-		}
-		char *type_name = NULL;
-		if (type_info->get_print_type) {
-			type_info->get_print_type(type_info, &type_name);
-		}
-		printf("  %s %s; //offset +0x%" PFMT64x "\n", type_name, member_name, offset);
-		RZ_FREE(type_name);
-	}
-	printf("};\n");
+	char *str = strdup(rz_strbuf_get(buf));
+	rz_strbuf_free(buf);
+	return str;
 }
 
-/**
- * \brief Prints out enum leaf type
- * 
- * \param name Name of the enum
- * \param type type of the enum
- * \param members List of cases
- * \param printf Print function
- */
-static void print_enum(const char *name, const char *type, const RzList *members, PrintfCallback printf) {
-	rz_return_if_fail(name && printf);
-	printf("enum %s { // type: %s\n", name, type);
-
-	RzListIter *member_iter;
-	STypeInfo *type_info;
-	rz_list_foreach (members, member_iter, type_info) {
-		char *member_name = NULL;
-		if (type_info->get_name) {
-			member_name = type_info->get_name(type_info);
-		}
-		ut64 value = 0;
-		if (type_info->get_val) {
-			value = type_info->get_val(type_info);
-		}
-		printf("  %s = %" PFMT64u ",\n", member_name, value);
-	}
-	printf("};\n");
-}
+static char *pdb_type_as_string_json(const RzTypeDB *db, const RzPdb *pdb, const RzList *types) {
```

Comment:
also this one I think it's all dupped code we don't need. Printing should be done by RzCore, not by RzBin or by PDB code. Moreover, the type can be easily printed by the type-specific code, no need to have it here in pdb code.

Reply:
As said above, the type of it is `RzBaseType`